lib/typescript/httpclient/api/airyConfig.ts

```diff
@@ -34,10 +34,16 @@ export const doFetchFromBackend = async (url: string, body?: Object): Promise<an
 
 async function parseBody(response: Response): Promise<any> {
   if (response.ok) {
-    return response.json();
+    try {
+      let jsonResponse = await response.json();
```

Comment:
Could you explain a bit the reasoning behind this change? Now it seems like clients of this function (so everywhere we call `doFetchFromBackend`) get `{}` in the unlikely case that the API returns non-json data for a successful call.

I get the idea of making it more robust, but I would suggest that we can just remove the `return` so that the error code further down in the function throws a proper error. What do you think?

Reply:
The idea behind this was that for successful calls that return non-json data, we wouldn't want to return an error and we wouldn't be able to consume the response again, so this why I returned {} (more like a temporary solution). 

I see your point and the problem of just returning {} in all cases, and I like the idea of removing the return in catch to let the function throw a proper error, as it would give us more flexibility. I guess we would automatically get an error for trying to call .text() on the response.